packages/asset-swapper/src/utils/quote_requestor.ts

```diff
@@ -0,0 +1,100 @@
+        // an AssetData object, which doesn't necessarily contain a
+        // token address.  (it could possibly be a StaticCallAssetData,
+        // which lacks an address.)  so we'll just assume it's a token
+        // here.  should be safe, with the enclosing guard condition
+        // and subsequent error.
+        // tslint:disable-next-line:no-unnecessary-type-assertion
+        return (decodedAssetData as ERC20AssetData).tokenAddress;
+    } else {
+        throw new Error(`Decoded asset data (${JSON.stringify(decodedAssetData)}) does not contain a token address`);
+    }
+}
+
+export class QuoteRequestor {
+    private readonly _rfqtMakerEndpoints: string[];
+
+    constructor(rfqtMakerEndpoints: string[]) {
+        this._rfqtMakerEndpoints = rfqtMakerEndpoints;
+    }
+
+    public async requestRfqtFirmQuotesAsync(
+        makerAssetData: string,
+        takerAssetData: string,
+        assetFillAmount: BigNumber,
+        marketOperation: MarketOperation,
+        takerApiKey: string,
+        takerAddress: string,
+        options?: Partial<RfqtFirmQuoteRequestOpts>,
+    ): Promise<SignedOrder[]> {
+        const { makerEndpointMaxResponseTimeMs } = _.merge({}, constants.DEFAULT_RFQT_FIRM_QUOTE_REQUEST_OPTS, options);
+
+        const buyToken = getTokenAddressOrThrow(makerAssetData);
+        const sellToken = getTokenAddressOrThrow(takerAssetData);
+
+        // create an array of promises for quote responses, using "undefined"
+        // as a placeholder for failed requests.
+
+        const responsesIfDefined: Array<void | AxiosResponse<SignedOrder>> = await Promise.all(
+            this._rfqtMakerEndpoints.map(async rfqtMakerEndpoint => {
+                try {
+                    return await Axios.get<SignedOrder>(`${rfqtMakerEndpoint}/quote`, {
+                        headers: { '0x-api-key': takerApiKey },
+                        params: {
+                            sellToken,
+                            buyToken,
+                            buyAmount: marketOperation === MarketOperation.Buy ? assetFillAmount.toString() : undefined,
+                            sellAmount:
+                                marketOperation === MarketOperation.Sell ? assetFillAmount.toString() : undefined,
+                            takerAddress,
+                        },
+                        timeout: makerEndpointMaxResponseTimeMs,
+                    });
+                } catch (err) {
+                    logUtils.warn(
+                        `Failed to get RFQ-T quote from market maker endpoint ${rfqtMakerEndpoint} for API key ${takerApiKey} for taker address ${takerAddress}: ${JSON.stringify(
+                            err,
+                        )}`,
+                    );
+                    return undefined;
+                }
+            }),
+        );
+
+        const responses: Array<AxiosResponse<SignedOrder>> = responsesIfDefined.filter(
```

Comment:
Can we use https://lodash.com/docs/4.17.15#compact here?

Reply:
I cleaned this up a _little_ bit.  See https://github.com/0xProject/0x-monorepo/pull/2541/commits/b854fcdb72f619a0a406fccf127547a7e9e15084.  What do you think?

In general I'm loath to introduce a 3rd-party lib to do something the language can handle, but if you feel strongly about it, or if there's some established convention around it, I'm happy to defer.

Reply:
Given that lodash is already a dependency of 0x-api, it feels natural to leverage it here. That being said, I'm ok with the code as-is